mac/build.sh

```diff
@@ -445,43 +244,88 @@ if $PREPRELEASE || $NOTARIZE; then
     echo
     builder_heading "Attempting to staple notarization to Keyman.app"
     xcrun stapler staple "$TARGET_APP_PATH" || builder_die "stapler failed"
+
+    DID_NOTARIZE=true
   fi
-fi
+}
 
-### Deploy as requested ###
-
-if $LOCALDEPLOY ; then
-    builder_heading "Attempting local deployment with command:"
-    KM4MIM_APP_BASE_PATH="$KM4MIM_BASE_PATH/build/$CONFIG"
-    displayInfo "$KM4MIM_BASE_PATH/localdeploy.sh \"$KM4MIM_APP_BASE_PATH\""
-    eval "$KM4MIM_BASE_PATH/localdeploy.sh" "$KM4MIM_APP_BASE_PATH"
-    if [ $? == 0 ]; then
-        displayInfo "Local deployment succeeded!" ""
-    else
-        builder_die "Local deployment failed!"
-    fi
-elif $PREPRELEASE ; then
-    builder_heading "Preparing files for release deployment..."
-    # Create the disk image
-    pushd setup
-    eval "./build.sh"
-    popd
-
-    eval "$KM4MIM_BASE_PATH/make-km-dmg.sh" $QUIET_FLAG
-    if [ $? == 0 ]; then
-        displayInfo "Creating disk image succeeded!" ""
-    else
-        builder_die "Creating disk image failed!"
-    fi
+do_sentry() {
+  # Upload symbols
+  builder_heading "Uploading symbols to sentry.keyman.com"
+
+  if which sentry-cli >/dev/null; then
+    pushd "$KM4MIM_BASE_PATH" > /dev/null
+    sentry-cli upload-dif "build/$CONFIG"
+
+    popd > /dev/null
+  else
+    builder_die "Error: sentry-cli not installed, download from https://github.com/getsentry/sentry-cli/releases"
+  fi
+
+  builder_finish_action success publish
+}
 
-    # Create download info
-    eval "$KM4MIM_BASE_PATH/write-download_info.sh"
-    if [ $? == 0 ]; then
-        displayInfo "Writing download_info file succeeded!" ""
-    else
-        builder_die "Writing download_info file failed!"
+do_install() {
+  if ! builder_has_option --quick; then
+    do_notarize
+  else
+    if [ "$(spctl --status)" == "assessments enabled" ]; then
+      echo
+      builder_warn "WARNING: Notarization is disabled but SecAssessment security policy is still active. Keyman will not run correctly."
+      builder_warn "         Disable SecAssessment with 'sudo spctl --master-disable' (or do notarized builds)"
+      builder_die  "Re-run without $(builder_term --quick) or disable SecAssessment."
     fi
-fi
+  fi
+
+  builder_heading "Attempting local deployment with command:"
+  KM4MIM_APP_BASE_PATH="$KM4MIM_BASE_PATH/build/$CONFIG"
+  displayInfo "$KM4MIM_BASE_PATH/localdeploy.sh \"$KM4MIM_APP_BASE_PATH\""
+  eval "$KM4MIM_BASE_PATH/localdeploy.sh" "$KM4MIM_APP_BASE_PATH"
+  
+  # TODO:  Consider removal?  Or can we not auto-crash if the prior line fails?
+  if [ $? == 0 ]; then
+      displayInfo "Local deployment succeeded!"
+  else
+      builder_die "Local deployment failed!"
+  fi
```

Comment:
In case of failure, I don't think we'll ever here, will we? This is probably legacy from before `set -e`